Sources/BSON/BSONError.swift

```diff
@@ -42,6 +42,19 @@ public enum BSONError {
     }
 }
 
+/// Standardize the errors emitted by BSONValue initializers.
+internal func BSONDecoderError(
+    codingPath: [CodingKey] = [],
+    keyPath: [String],
+    debugDescription: String
+) -> DecodingError {
+    let debugStart = keyPath.joined(separator: ".")
+    return DecodingError.dataCorrupted(DecodingError.Context(
+        codingPath: codingPath,
+        debugDescription: debugStart + debugDescription
```

Comment:
as written, this will produce strings like "x.yExpected only \"$numberInt\" key, found extra keys in object: []", there needs to be a colon and space here. For cases where the key path is empty, make sure to omit them though, otherwise we'll have ": <rest of error message>".

Reply:
fixed